ThermoFun/Database.cpp

```diff
@@ -114,34 +114,34 @@ struct Database::Impl
         reactions_map = reactions;
     }
 
-    auto getElements() -> std::vector<Element>
+    auto getElements() const -> std::vector<Element>
     {
         return collectValues(elements_map);
     }
 
-    auto getSubstances() -> std::vector<Substance>
+    auto getSubstances() const -> std::vector<Substance>
     {
         return collectValues(substances_map);
     }
 
-    auto getReactions() -> std::vector<Reaction>
+    auto getReactions() const -> std::vector<Reaction>
     {
         return collectValues(reactions_map);
     }
 
-    auto numberOfElements() -> size_t
+    auto numberOfElements() const -> size_t
     {
-        return collectValues(elements_map).size();
```

Comment:
We do not need to collect all elements to determine the size! :)
Unless there is something else going on here that I'm missing.

Reply:
yes you are totally right, I am confused myself about why it was like that.